src/main/java/uk/ac/cam/cl/dtg/segue/api/UsersFacade.java

```diff
@@ -709,30 +708,23 @@ private Response updateUserObject(final HttpServletRequest request, final HttpSe
                 }
             }
 
-            // Now update the email preferences
-            emailPreferenceManager.saveEmailPreferences(userObjectFromClient.getId(), emailPreferences);
+            if (userPreferences != null) {
 
-            //----------------------------------------------------------------------------------------------------------
-            // FIXME - the code between the dashed lines should be refactored out of this class; it does not belong here!
```

Comment:
Where has the validation code gone to? This was the only thing preventing people from adding random junk to the database directly . . .